lib/persister/game_data.rb

```diff
@@ -0,0 +1,130 @@
+require_relative '../game_mode/lite_3';
+
+module GameData
+    extend self
+
+    def serializer_types(game)
+        @game = game
+        types.map do |type|
+            type.delete(:deserialize)
+            type
+        end
+    end
+
+    def deserializer_types
+        types.map do |type|
+            type.delete(:data)
+            type
+        end
+    end
+
+    private
+    attr_accessor :game
+
+    def types
+        [
+            board_presenter,
+            status,
+            game_mode,
+            board_positions,
+            board_size,
+            current_player_token,
+            player_2,
+            player_1_moves,
+            player_2_moves
+        ]
+    end
+
+    def board_presenter
+        {
+            name: :board_presenter,
+            data: game && game.board_presenter.presenter.class.name
+        }
+    end
+
+    def status
+        {
+            name: :status,
+            data: game && game.status,
+            deserialize: -> (string) { status = convert_to_symbol(string); status = status == :save ? :play : status  }
+        }
+    end
+
+    def game_mode
+        {
+            name: :game_mode,
+            data: game && game.game_state.game_mode.type,
```

Comment:
If you find yourself reaching really far into an object to retrieve some value, it is often considered a code smell. In this case, game seems like some kind of god class that knows way too much about the internals of its subcomponents. 

Refer: https://en.wikipedia.org/wiki/Law_of_Demeter

Reply:
I did think I was toeing the line with how far I was reaching -- but I couldn't think of another way of accessing all of this info other than to have getter fns that stores the data.

for example, game_state would then keep track of the game_mode type, etc.

this was something I'd like to discuss more with you guys during code review :D

Reply:
This might also be helpful reading https://dzone.com/articles/the-genius-of-the-law-of-demeter